CHANGELOG.md

```diff
@@ -7,9 +7,12 @@
 - **BREAKING**: For `TimeoutError`, change the error message to include the
   request's url. Was: `Request timed out`, 
   now: `Request to "example.com" timed out.`
+- **BREAKING**: Remove default exports.
 
 ### Added
 - Add a `.requestUrl` property to all errors, for easier logging/diagnostics.
+- Add `PUT` to allowed proxied methods.
+- Add `customClient()` factory method (useful to override default headers etc).
```

Comment:
As an alternative to this could we just expose `create` on the proxy as well -- to keep the API as similar as possible to `ky`? In fact, is it not already exposed -- have we tried just calling `httpClient.create()`?

Reply:
This is aligned with the spirit of this library, a thin, almost transparent wrapper around `ky`. Additional exports should only be added if absolutely necessary. One should look at the `ky` documentation first to see if the desired feature exists. It seems that `ky.extend` could be of use here. See https://github.com/sindresorhus/ky#kyextenddefaultoptions

Reply:
@dlongley @gannan08 - I'm open to suggestions. 
I couldn't figure out how to use `ky.extend` and keep the proxy instance. In tests, calling the (proxied) `httpClient.extend()` (same with `.create()`) resulted in what looked like a non-proxied client, without the default headers and the expected behavior.

Reply:
@dmitrizagidulin, what I was thinking was that the proxy would override the `create` and/or `extend` methods to return a proxied version of whatever those return.

Reply:
@dlongley - ah, I see. that should be easy enough.

Reply:
@dlongley @gannan08 -- refactored. New usage is:

```js
import {httpClient} from `@digitalbazaar/http-client`;

const headers = {Authorization: `Bearer ${accessToken}`};
const client = httpClient.extend({headers, httpsAgent});
```